File: backend/internal/cli/system.go
package cli

import "context"

func (e commandEnv) runSystem(ctx context.Context, args []string) int {
	if len(args) == 0 {
		return e.fail("missing system command")
	}
	if args[0] == "status" || args[0] == "doctor" {
		return e.getSystemStatus(ctx, args[1:])
	}
	return e.fail("unknown system command: " + args[0])
}

func (e commandEnv) getSystemStatus(ctx context.Context, args []string) int {
	fs := newFlagSet("system status")
	addr, asJSON := e.addCommonFlags(fs)
	if err := fs.Parse(args); err != nil {
		return e.fail(err.Error())
	}
	ctx = withFlagToken(ctx, fs)
	var status systemStatusDTO
	if err := getJSON(ctx, e.client, *addr, "/api/system/status", &status); err != nil {
		return e.fail(err.Error())
	}
	if err := writeSystemStatus(e.out, status, *asJSON); err != nil {
		return e.fail(err.Error())
	}
	return 0
}

File: backend-ts/src/cli/system.ts
import { parseCommandArgs } from "./common.ts";
import { getJSON } from "./http.ts";
import { formatSystemStatus } from "./output.ts";
import type { EnvReader, Fetcher, SystemStatusDTO } from "./types.ts";

export async function runSystem(args: string[], env: EnvReader, fetcher: Fetcher): Promise<string> {
  const command = args[0]?.trim();
  if (!command) throw new Error("missing system command");
  if (command !== "status" && command !== "doctor") throw new Error(`unknown system command: ${command}`);
  return await getSystemStatus(args.slice(1), env, fetcher);
}

export async function getSystemStatus(args: string[], env: EnvReader, fetcher: Fetcher): Promise<string> {
  const { common } = parseCommandArgs(args, [], env);
  const status = await getJSON<SystemStatusDTO>(fetcher, common, "/api/system/status");
  return formatSystemStatus(status, common.json);
}
